src/main/java/org/mule/extension/http/api/request/builder/HttpRequesterSimpleRequestBuilder.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.http.api.request.builder;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Collections.unmodifiableList;
+import static org.mule.extension.http.internal.request.KeyValuePairUtils.toMultiMap;
+import static org.mule.runtime.http.api.server.HttpServerProperties.PRESERVE_HEADER_CASE;
+
+import org.mule.extension.http.internal.request.HttpRequesterConfig;
+import org.mule.extension.http.internal.request.UriUtils;
+import org.mule.runtime.api.metadata.TypedValue;
+import org.mule.runtime.extension.api.annotation.param.NullSafe;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.annotation.param.display.DisplayName;
+import org.mule.runtime.extension.api.annotation.param.display.Text;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.request.HttpRequestBuilder;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Component that specifies how to create a proper HTTP simple request, that can be used in a Source
+ * and connectivity testing.
+ *
+ * @since 1.7
+ */
+public class HttpRequesterSimpleRequestBuilder {
+
+  /**
+   * The body of the response message
+   */
+  @Parameter
+  @Optional(defaultValue = "")
+  @Text
+  @DisplayName("Body")
+  private String requestBody;
+
+  /**
+   * HTTP headers the message should include.
+   */
+  @Parameter
+  @Optional
+  @NullSafe
+  @DisplayName("Headers")
+  protected List<SimpleRequestHeader> requestHeaders = emptyList();
+
+  /**
+   * URI parameters that should be used to create the request.
+   */
+  @Parameter
+  @Optional
+  @DisplayName("URI Parameters")
+  private Map<String, String> requestUriParams = emptyMap();
+
+  /**
+   * Query parameters the request should include.
+   */
+  @Parameter
+  @Optional
+  @DisplayName("Query Parameters")
+  private List<QueryParam> requestQueryParams = emptyList();
+
+  public TypedValue<Object> getRequestBody() {
+    return TypedValue.of(requestBody);
+  }
+
+  public void setRequestBody(String body) {
```

Comment:
are the setters really needed? Best practice is for these classes to be immutable (or as close as we can get to that)

Reply:
The problem is potentially doing some unit testing (for the connectivity testing or polling source) without being able to set this manually

Reply:
then I would only add them as protected when needed, specially since this is API